app/services/supplementary-billing/abstraction-billing-period.service.js

```diff
@@ -137,23 +139,61 @@ function _abstractionPeriods (billingPeriod, chargePurpose) {
 }
 
 /**
- * Calculates the 'billable days' for each abstraction billing period
+ * Calculates the 'billable days' for each abstraction billable period
  *
  * @param {Object[]} abstractionPeriods An array of abstraction billing periods
  *
  * @returns {Object[]} The array abstraction periods each with a new `billableDays` property
  * @returns {Date} periods[].startDate
  * @returns {Date} periods[].endDate
  * @returns {boolean} periods[].consider
+ * @returns {Date} periods[].billableStartDate
+ * @returns {Date} periods[].billableEndDate
  * @returns {number} periods[].billableDays
  */
 function _calculateBillableDays (abstractionPeriods) {
-  let difference
-  let billableDays
   for (const abstractionPeriod of abstractionPeriods) {
-    difference = abstractionPeriod.endDate.getTime() - abstractionPeriod.startDate.getTime() // difference in msecs
-    billableDays = Math.ceil(difference / (1000 * 3600 * 24)) + 1 // (1000 msecs * (60 secs * 60 mins) * 24 hrs)
-    abstractionPeriod.billableDays = billableDays
+    if (abstractionPeriod.billableStartDate) {
+      const difference = abstractionPeriod.billableEndDate.getTime() - abstractionPeriod.billableStartDate.getTime() // difference in msecs
+      const billableDays = Math.ceil(difference / (1000 * 3600 * 24)) + 1 // (1000 msecs * (60 secs * 60 mins) * 24 hrs)
```

Comment:
Just for clarity is it worth maybe putting the `24 * 60 * 60 * 1000` calculation as a constant defined at the top, eg. `DAYS_IN_MILLISECONDS` or something like that? And maybe stick a comment in here to explain what the `billableDays` calculation is doing?

Reply:
I shall sort that out